13 switch to env

JackDunnCodes opened the pull request.

JackDunnCodes committed:
Bye-bye config.json, hello .env

JackDunnCodes committed:
First pass on replacing config.env

JackDunnCodes committed:
Second pass, replacing config.env

JackDunnCodes committed:
Update CI

JackDunnCodes committed:
Match Next's expectations for .env

JackDunnCodes committed:
Switch to react-env, for runtime env
I didn't want this to be so complicated but here we are

JackDunnCodes committed:
Improve(?) dockerfile

JackDunnCodes committed:
Dockerfile fix

JackDunnCodes committed:
Run react-env before server start, not before build

JackDunnCodes committed:
typo

JackDunnCodes commented:
closes #13

SuperNinja-4965 commented on .github/workflows/Build-Test-Code.yml:
```diff
@@ -64,6 +64,5 @@ jobs:
     - run: npm install --global yarn
     - run: cd $GITHUB_WORKSPACE && yarn install
     # Run the build
-    - run: cp $GITHUB_WORKSPACE/config-example.json $GITHUB_WORKSPACE/config.json
```
This line needs to be replaced with `cp $GITHUB_WORKSPACE/.env.template $GITHUB_WORKSPACE/.env` like you have on the tests does it not?

JackDunnCodes replied:
Since the .env will be included at runtime, not at build, it doesn't need the config here - as far as I'm aware it'll only be needed for tests

SuperNinja-4965 replied:
ok. the reason i asked is because for the config.json the build fails without it.

JackDunnCodes replied:
Makes sense - we don't want to include the config at build because then the docker image is permanently locked to that config and we'd also probably accidentally end up publishing API keys which is certified Not A Good Idea™

SuperNinja-4965 committed:
Update yarn clean to exclude .env instead of config.json

SuperNinja-4965 committed:
Autobuild and Publish Docker Image to Docker Hub (#16)

* initial work

* Add Action to build docker image.
Needs properly testing but in theory this should detect if this is a tag or not and then upload the built image accordingly

* Fixed stable and beta id clash

* Updated to so branches can be identified and uploaded separately

Co-authored-by: Jack Dunn <56315967+JackDunnCodes@users.noreply.github.com>

JackDunnCodes committed:
Fix docker build

JackDunnCodes commented:
now closes #15

SuperNinja-4965 commented:
just hold off one second on merging

JackDunnCodes commented:
> just hold off one second

i ain't closing yet, just tagging :)

SuperNinja-4965 committed:
Remove old pretest and update pretest to just call copy.

SuperNinja-4965 reviewed (approved):
lgtm
